tests/features/S2N_LIBCRYPTO_SUPPORTS_KYBER.c

```diff
@@ -16,12 +16,15 @@
 #include <openssl/evp.h>
 #include <openssl/nid.h>
 
-int main() {
+int main()
+{
     EVP_PKEY_CTX *ctx = EVP_PKEY_CTX_new_id(EVP_PKEY_KEM, NULL);
     if (ctx == NULL) {
         return 1;
     }
-    if (!EVP_PKEY_CTX_kem_set_params(ctx, NID_KYBER512_R3)) {
+    if (!EVP_PKEY_CTX_kem_set_params(ctx, NID_KYBER512_R3)
+            || !EVP_PKEY_CTX_kem_set_params(ctx, NID_KYBER768_R3)
+            || !EVP_PKEY_CTX_kem_set_params(ctx, NID_KYBER1024_R3)) {
```

Comment:
Wouldn't this potentially break existing users of Kyber? Currently, they can use Kyber if the EVP_PKEY_CTX_kem_set_params and NID_KYBER512_R3 symbols exist. Now you're also requiring the NID_KYBER768_R3 and NID_KYBER1024_R3 symbols, which might not be in their version of awslc. So if they update s2n-tls without updating awslc, they could break.

Reply:
> Currently, they can use Kyber if the EVP_PKEY_CTX_kem_set_params and NID_KYBER512_R3 symbols exist. Now you're also requiring the NID_KYBER768_R3 and NID_KYBER1024_R3 symbols, which might not be in their version of awslc

`NID_KYBER512_R3` was added in [this commit](https://github.com/aws/aws-lc/commit/ee63cde2d128398bd0632414d120fb0f15841c5a) and released in [v1.4.0](https://github.com/aws/aws-lc/releases/tag/v1.4.0) on 2/7/23. `NID_KYBER768_R3` and `NID_KYBER1024_R3` were added in [this commit](https://github.com/aws/aws-lc/commit/fb42ac63b765d1be7bb8cb5537bbd31c6c906976) and released in [v1.5.0](https://github.com/aws/aws-lc/releases/tag/v1.5.0) 3 weeks later on 3/1/23. So, this would pertain to consumers who've pinned AWS-LC to v1.4.0 but continue to update s2n-tls.

> So if they update s2n-tls without updating awslc, they could break.

In this scenario, the feature probe would fail, s2n-tls would fall back to its [interned Kyber512 implementation](https://github.com/aws/s2n-tls/blob/main/pq-crypto/kyber_r3/kyber512r3_kem.c), and (importantly) not attempt to compile Kyber768-dependent code. I believe this is the behavior we want, and wouldn't result in breakage.

> You could also simplify this feature test, since it doesn't really matter what it returns because it never gets run, just compiled.

Yes, but I vaguely remember one of the static analysis tools in the CI yelling at me the last time I touched this for not handling errors or properly freeing.

Reply:
To confirm this, I checked out, built, and installed AWS-LC v1.4.0 locally. I then pointed s2n's build to that and ran the unit tests against this PR's changeset. All tests were successful.

Reply:
Duh, I totally forgot there was an interned implementation to fallback to :) Makes sense.